model/QtiJsonItemCompiler.php

```diff
@@ -253,4 +262,54 @@ private function getItemAssetXmlReplacer(): QtiItemAssetXmlReplacer
     {
         return $this->getServiceLocator()->get(QtiItemAssetXmlReplacer::class);
     }
+
+    private function getEnhancedMetadataProperties(): array
+    {
+        $triples = $this->getResource()->getRdfTriples();
+        $properties = [];
+        $metaDataKeysContext = [];
+        $metaDataValuesContext = [];
+        foreach ($triples as $triple) {
+            if (!empty($properties[$triple->predicate])) {
+                $properties[$triple->predicate] = [$properties[$triple->predicate], $triple->object];
+            } else {
+                $properties[$triple->predicate] = $triple->object;
+            }
+            $metaDataKeysContext[] = $triple->predicate;
+            if (strpos($triple->object, 'http') !== false) {
+                $metaDataValuesContext[] = $triple->object;
+            }
+        }
+        //we also include a shortcut to the item URI
+        $properties['metaDataKeysContext'] = $this->retrieveMetaDataContext(array_unique($metaDataKeysContext));
+        $properties['metaDataValuesContext'] = $this->retrieveMetaDataContext(array_unique($metaDataValuesContext));
+        $properties['@uri'] = $this->getResource()->getUri();
+
+        return $properties;
+    }
+
+    private function retrieveMetaDataContext(array $array): iterable
+    {
+        $newArray = [];
+        foreach ($array as $key) {
+            $resource = $this->getResourceLabelByIdentifier($key);
+            $newArray[] = [
+                '@id' => $key,
+                '@label' => !is_null($resource) ? $resource->getLabel() : null
+            ];
+        }
+
+        return $newArray;
+    }
+
+    private function getResourceLabelByIdentifier(string $uri): ?core_kernel_classes_Resource
```

Comment:
You are not getting label here, but the resource. Suggest change name or extend method. Btw, why not use `OntologyAwareTrait` though?

```suggestion
    private function getResourceByIdentifier(string $uri): ?core_kernel_classes_Resource
```